azavea/azavea-branding-guide
# Add search functionality

## Overview
This PR adds search functionality to the branding guide without using a plugin.

### Demo
<img width="738" alt="screen shot 2018-08-15 at 3 44 41 pm" src="https://user-images.githubusercontent.com/5672295/44169409-55d57a00-a0a2-11e8-8030-314b8ac0da25.png">
<img width="539" alt="screen shot 2018-08-15 at 3 44 44 pm" src="https://user-images.githubusercontent.com/5672295/44169410-55d57a00-a0a2-11e8-8128-157e03174958.png">
<img width="263" alt="screen shot 2018-08-15 at 3 44 53 pm" src="https://user-images.githubusercontent.com/5672295/44169412-55d57a00-a0a2-11e8-8ddd-2b3e3256ec80.png">


### Notes
- The repo that this pulls from is now archived, but it does seem to work so lemme know if that's a problem: https://github.com/jekylltools/jekyll-tipue-search
- I hid the search on mobile just to save screen space, but I can try and figure out something different if we should try and retain this for mobile users.
- It doesn't seem like I can change the text created in `tipuesearch_set.js`. I only really wanted to change "Back" and "More" in the pagination to icons, which I thought might be more intuitive, but doing so seemed to break the search functionality.

## Testing Instructions
* `git pull`
* Run a search for "nothin" to get the "Nothing found message"
* Run a search for "blog" to see results. Result headers should be clickable and take you to the proper page.
* Run an empty search to see the warning messages of "Search too short" and "Should be 3 characters or more"
* Run a search for "azavea" to see pagination.

Closes #1
